test/e2e/cypress/integration/profiles/recovery/code/errors.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { extractRecoveryCode, appPrefix, gen } from '../../../../helpers'
+import { routes as react } from '../../../../helpers/react'
+import { routes as express } from '../../../../helpers/express'
+
+context('Account Recovery Errors', () => {
+  ;[
+    {
+      recovery: react.recovery,
+      app: 'react' as 'react',
+      profile: 'spa'
+    },
+    {
+      recovery: express.recovery,
+      app: 'express' as 'express',
+      profile: 'recovery'
+    }
+  ].forEach(({ recovery, profile, app }) => {
+    describe(`for app ${app}`, () => {
+      before(() => {
+        cy.deleteMail()
+        cy.useConfigProfile(profile)
+        cy.proxy(app)
+      })
+
+      beforeEach(() => {
+        cy.deleteMail()
+        cy.longRecoveryLifespan()
+        cy.longCodeLifespan()
+        cy.disableVerification()
+        cy.enableRecovery('code')
+      })
+
+      it('shows code expired message if expired code is submitted', () => {
```

Comment:
Please cover the "tried too often" use case!

Reply:
Done.